fence/auth.py

```diff
@@ -39,6 +43,78 @@ def get_jwt():
         raise Unauthorized("expected bearer token in auth header")
     return token
 
+def requested_by_gen3_service():
+    """
+    Returns the contents of the Gen3-Service header or False
+    """
+    header_utf8 = flask.request.headers.get("Gen3-Service", '')
+    if header_utf8:
+        header = header_utf8.decode('utf-8')
+        logger.info(f"Fence auth requested_by_gen3_service header: {header}")
+        return header
+    
+    return False
+
+def request_is_gen3_signed():
+    """
+    Returns bool if Signature header is present
+    """
+    header_utf8 = flask.request.headers.get("Signature", '')
+    header = header_utf8.decode('utf-8')
+    logger.info(f"Fence auth request_is_gen3_signed header: {header}")
+    return (bool(header) and bool(requested_by_gen3_service()))
+
+def get_request_signature():
+    """
+    Return the signature header from another service. Requires flask application context.
+    Raises:
+        - Unauthorized, if header is missing or not in the correct format
+    """
+    header = flask.request.headers.get("Signature", None)
+    if not header:
+        raise Unauthorized("missing signature header")
+    try:
+        signature_string = header.decode('utf-8')
+        prefix, hexed_signature = signature_string.split(" ")
+        signature = binascii.unhexlify(hexed_signature)
+    except ValueError:
+        raise Unauthorized("signature header not in expected format")
+    if prefix.lower() != "signature":
+        raise Unauthorized("expected signature token in auth header")
+    return signature
+
+def verify_request_signature(public_key_path, data, signature):
+    """
+    Check if the signature header is valid
+    """
+    if not public_key_path and not signature and not data:
+        return False
+    try:
+        pub_key = open(public_key_path, "r").read()
+        rsakey = RSA.importKey(pub_key)
```

Comment:
Instead of reading at every request the path from the env variable, open the file and import the key, I would do all these actions at startup and load the already imported key in the flask config. In this way we can have a faster request.